### PR TITLE
Feat/batch optimization

### PR DESCRIPTION
**Pull request by oumar-fall**
### Pull-Request Checklist

- [ ] Code is up-to-date with the `main` branch
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: src/mocks/global.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import { GlobalData } from "../types";
 import { Underlying } from "./markets";
 
 const SECONDS_PER_WEEK = 24 * 3600 * 7; // in s
+export const BASE_BLOCK_TIMESTAMP = 1679584232;
```
Comment (julien-devatom) [inline src/mocks/global.ts:10]:
```suggestion
export const INITIAL_BLOCK_TIMESTAMP = 1679584232;
```

base is a chain

File: src/txHandler/Bulker.TxHandler.ts [hunk 2]
```diff
@@ -144,9 +144,139 @@ export default class BulkerTxHandler
     await new Promise((resolve) => {
       this.#done$?.subscribe(resolve);
 
-      const operations = this.simulatorOperations$.getValue();
+      let operations = this.simulatorOperations$.getValue();
 
-      this.simulatorOperations$.next([...operations, operation]);
+      let newOperation: Operation | undefined | null;
```
Comment (julien-devatom) [inline src/txHandler/Bulker.TxHandler.ts:149]:
```suggestion
      let newOperation?: Operation | null;
```

File: tests/units/txBatch/bulker.optimisation.test.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+              relation: OperationRelation.equivalent,
+            },
+            {
+              txType: baseTxType,
+              reverse: true,
+              max: true,
+              relation: OperationRelation.bigger,
+            },
+          ].forEach(({ txType, reverse, max, relation }) => {
+            it(`with a ${relation ? relation + " " : ""}${getTxType(
```
Comment (julien-devatom) [inline tests/units/txBatch/bulker.optimisation.test.ts:235]: you can leverage on a `describe.each` for a better formatting of the tests

File: tests/units/txBatch/bulker.optimisation.test.ts [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+  });
+
+  [
+    TransactionType.supply,
+    TransactionType.borrow,
+    TransactionType.supplyCollateral,
+  ].forEach((baseTxSide) => {
+    [false, true].forEach((baseReverse) => {
+      [true, false].forEach((baseMax) => {
+        describe(`a ${getTxType(baseTxSide, baseReverse)}${
```
Comment (julien-devatom) [inline tests/units/txBatch/bulker.optimisation.test.ts:180]: you can leverage on a `describe.each` for a better formatting of the tests

**Review by julien-devatom — COMMENTED** [review]
need a rebase

File: tests/units/txBatch/bulker.optimisation.test.ts (hunk 3 above)
Comment (oumar-fall) [reply tests/units/txBatch/bulker.optimisation.test.ts:235]: As far as I remember I tried but i had some limitations

File: tests/units/txBatch/bulker.optimisation.test.ts (hunk 4 above)
Comment (oumar-fall) [reply tests/units/txBatch/bulker.optimisation.test.ts:180]: As far as I remember I tried but i had some limitations

File: src/txHandler/Bulker.TxHandler.ts (hunk 2 above)
Comment (oumar-fall) [reply src/txHandler/Bulker.TxHandler.ts]: That's not a valid ts syntax lol

**Review by julien-devatom — APPROVED** [review]

**Pull request merged**